spec/truffle/identity_spec.rb

```diff
@@ -8,25 +8,81 @@
 
 require_relative '../ruby/spec_helper'
 
-# Require something that uses code from RubySL
-require 'stringio'
+require 'date'
+require 'rubygems'
 
 describe "Identifying features such as" do
 
-    describe "the Rubinius module" do
+  it "RUBY_ENGINE is 'truffleruby'" do
+    RUBY_ENGINE.should == 'truffleruby'
+  end
 
-      it "is not defined" do
-        defined?(Rubinius).should be_nil
-      end
-      
+  it "RUBY_VERSION matches /\d+\.\d+\.\d+/" do
+    RUBY_VERSION.should =~ /\d+\.\d+\.\d+/
+  end
+
+  it "RUBY_ENGINE_VERSION matches /\d+\.\d+" do
+    RUBY_ENGINE_VERSION.should =~ /\d+\.\d+/
+  end
+
+  it "RUBY_REVISION is 0" do
+    RUBY_REVISION.should == 0
+  end
+
+  it "RUBY_PATCHLEVEL is 0" do
+    RUBY_PATCHLEVEL.should == 0
+  end
+
+  it "RUBY_RELEASE_DATE is the current year" do
+    RUBY_RELEASE_DATE.should == Date.today.year.to_s
+  end
+
+  it "Gem.ruby_engine matches RUBY_ENGINE" do
+    Gem.ruby_engine.should == RUBY_ENGINE
+  end
+
+  it "Gem.ruby_version matches RUBY_VERSION as a Gem version object" do
+    Gem.ruby_version.should == Gem::Version.new(RUBY_VERSION)
+  end
+
+  it "RbConfig::CONFIG['ruby_install_name'] is 'truffleruby'" do
```

Comment:
In the description, I would simplify like `ruby_install_name is 'truffleruby'`

Reply:
I don't agree - I think explicit is better here.

Reply:
OK.